lib/discovery-indexer/reader/purlxml_parser_strict.rb

```diff
@@ -165,13 +183,12 @@ def parse_use_and_reproduction
       # the @id attribute of resource/file elements, including extension
       # @return [Array<String>] filenames
       def parse_file_ids
-        ids = []
         content_md = parse_content_metadata
-        return nil if content_md.nil?
+        return [] if content_md.nil?
+        ids = []
         content_md.xpath('//resource/file/@id').each do |node|
```

Comment:
If we're going w/ the `.map` solution above in might make sense to do so here as well (so we can get rid of the `ids` temp variable and just have the `.map` be the return value of the method).  `.map{}.compact` will return an empty array if the only values are `nil`.